src/loki/worker.cc

```diff
@@ -83,13 +84,26 @@ void loki_worker_t::parse_costing(Api& api, bool allow_none) {
     }
   } catch (const std::runtime_error&) { throw valhalla_exception_t{125, "'" + costing_str + "'"}; }
 
-  // See if we have avoids and take care of them
-  if (static_cast<size_t>(options.avoid_locations_size()) > max_avoid_locations) {
-    throw valhalla_exception_t{157, std::to_string(max_avoid_locations)};
+  if (options.avoid_polygons_size()) {
+    // TODO: should the PBF edge assignment happen in edges_in_rings?
+    // feels like a waste to produce an unordered_set we just iterate over right after
```

Comment:
i agree with the sentiment here but theres nothing else we can do, protobuf doesnt support hashsets natively and we kind of need the deduplication when we are doing the intersection. seems theres not a good clean solution here. i thikn what you have is reasonable tbh